psycopg/psycopg/connection.py

```diff
@@ -271,11 +271,17 @@ def cancel_safe(self) -> None:
         If the underlying libpq is older than version 17, the method will fall
         back to using the same implementation of `!cancel()`.
         """
-        if self._should_cancel():
+        if not self._should_cancel():
+            return
+
+        # TODO: replace with capabilities.has_safe_cancel after merging #782
+        if pq.__build_version__ >= 170000:
             try:
                 waiting.wait_conn(self._cancel_gen(), interval=_WAIT_INTERVAL)
-            except e.NotSupportedError:
-                self.cancel()
+            except Exception as ex:
+                logger.warning("couldn't try to cancel query: %s", str(ex))
```

Comment:
```suggestion
                logger.warning("couldn't try to cancel query: %s", ex)
```

Reply:
I started to use this pattern since I discovered that Sentry might leak secrets in its default configuration (https://github.com/getsentry/sentry-python/issues/2417#issuecomment-1752447375).

Writing this code actually made me think I'd I shouldn't actually protect every single rerause of exceptions of too broad classes.